src/main/java/org/jenkinsci/test/acceptance/docker/Docker.java

```diff
@@ -158,6 +125,107 @@ public DockerImage build(Class<? extends DockerContainer> fixture) throws IOExce
         }
     }
 
+    //package scope for testing purposes. Ideally we should encapsulate Docker interactions so they can be mocked
+    // and call public method.
+    void copyDockerfileDirectory(Class<? extends DockerContainer> fixture, DockerFixture f, File dir)
+            throws IOException {
+        String dockerfilePath = resolveDockerfileLocation(fixture, f);
+
+        if(dockerfilePath.startsWith(FILE_SCHEMA)) {
```

Comment:
I thought the same but then I open the door that someone could use any schema like `http` or `ftp` since Java provides out of the box this support. So I decided to put some protection on this.
But of course I could implement the `if`as now but then call `copyURLToFile` as you mention.